libs/openFrameworks/sound/ofMediaFoundationSoundPlayer.h

```diff
@@ -0,0 +1,276 @@
+#pragma once
+
+#include "ofSoundBaseTypes.h"
+#include "ofEvents.h"
+
+#include <mfidl.h>
+#include <mfapi.h>
+#include <mfreadwrite.h>
+#include <wrl.h>
+#include <xaudio2.h>
+
+#include <list>
+
+// https://github.com/microsoft/DirectXTK/blob/main/Audio/AudioEngine.cpp
+
+namespace of {
+	struct MFSourceReaderNotifyCallback {
+	public:
+		virtual void OnSourceReaderEvent(HRESULT hrStatus, DWORD dwStreamIndex,
+			DWORD dwStreamFlags, LONGLONG llTimestamp, IMFSample* pSample) = 0;
+	};
+}
+
+class ofMediaFoundationUtils {
```

Comment:
Moved some media foundation utils into the MF sound player class so that it could be shared with the Media Foundation video player.